### PR TITLE
feat: create-vuestic

### PR DESCRIPTION
**Pull request by m0ksem**
closes https://github.com/epicmaxco/vuestic-ui/issues/2204

CLI tool to create new app with vuestic-ui. Should replace vue-cli-plugin-vuestic. Must be recommended way to create Vuestic Admin app too.

```
npm create vuestic@latest
```
image

Three templates:
image

Prompts:
- TreeShaking (for vite & nuxt)
- AgGrid
- * Might add epicspinners
- * Might be tailwind

### CODE REVIEW

**Comment by aluarius** [comment]
Just in case (for nuxt):

image

**Comment by aluarius** [comment]
Think we should provide here more info (for dummies).

- Vue 3 + Vite
- Nuxt 3

image

**Comment by aluarius** [comment]
May be we should add test Vuestic UI components to the default App.vue? Other UI frameworks do. Instant demonstation.

**Comment by aluarius** [comment]
Does `treeshaking` option affects result for nuxt?

**Comment by aluarius** [comment]
image

**Comment by aluarius** [comment]
I'd prefer not to make preselected options, because it's not obvious - i've thought that green circle is a currently hovered option.

image

File: packages/create-vuestic/src/steps/2.addVuestic/vue-app.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export const addVuesticToVue3App = async () => {
+  const { projectName } = await useUserAnswers()
+
+  // Install vuestic-ui
+  const { addDependency } = await usePackageJson()
+  addDependency('vuestic-ui', 'latest')
+
+  // Add plugin
+  const mainPath = resolvePath(process.cwd(), projectName, 'src/main.js') || resolvePath(process.cwd(), projectName, 'src/main.ts')
```
Comment (aluarius) [inline packages/create-vuestic/src/steps/2.addVuestic/vue-app.ts:19]: Why not via resolve-path.ts?

File: packages/create-vuestic/src/steps/2.addVuestic/insert-nuxt-module.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  vuestic: {
+    ${vuesticConfig.split('\n').join('\n    ')}
+  },
+`
+}
+
+export const insertNuxtModule = (source: string, css: string[]) => {
+  const vuesticConfig = createVuesticConfig(css)
+
+  if (/vuestic: \{/.test(source) || /modules: \[((.|\n)*)\]/gm.test(source)) {
```
Comment (aluarius) [inline packages/create-vuestic/src/steps/2.addVuestic/insert-nuxt-module.ts:34]: Does this situation possible? 0_o

**Review by aluarius — APPROVED** [review]

File: packages/create-vuestic/src/steps/2.addVuestic/insert-nuxt-module.ts (hunk 2 above)
Comment (m0ksem) [reply packages/create-vuestic/src/steps/2.addVuestic/insert-nuxt-module.ts:34]: Not yet. But would be nice if create-vuestic can append vuestic-ui to already existing project in future.

File: packages/create-vuestic/src/generators/create-vue.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { execp } from './../utils/exacp';
+
+type CreateVueFeature = 'ts' | 'jsx' | 'router' | 'pinia' | 'tests' | 'vitest' | 'cypress' | 'playwright' | 'eslint'
```
[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/create-vuestic/src/generators/create-vue.ts:3]: I was thinking if we should copy create-vue behavior, but missing playwright makes this CLI easier, so I think to skip it for now. This type represents create-vue features, but we will ignore playwright (propose cypress instead). And I just forgot about JSX.

[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/create-vuestic/src/generators/create-vue.ts:3]: Having both playwright and cypress requires to have select between this two, but we have multipleselect of all create-vue features in one window.

[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/create-vuestic/src/generators/create-vue.ts:3]: I added playwright and jsx here.

File: packages/create-vuestic/src/steps/2.addVuestic/vue-app.ts (hunk 1 above)
Comment (m0ksem) [reply packages/create-vuestic/src/steps/2.addVuestic/vue-app.ts:19]: I use here resolvePath, no?

**Review by rustem-nasyrov — APPROVED** [review]

**Pull request merged**